src/metric/definition.go

```diff
@@ -689,15 +689,15 @@ var KSMSpecs = definition.SpecGroups{
 		TypeGenerator: prometheus.FromLabelValueEntityTypeGenerator("kube_pod_status_phase"),
 		Specs: []definition.Spec{
 			{Name: "createdAt", ValueFunc: prometheus.FromValue("kube_pod_created"), Type: sdkMetric.GAUGE, Optional: true},
-			{Name: "startTime", ValueFunc: prometheus.FromValue("kube_pod_start_time"), Type: sdkMetric.GAUGE},
 			{Name: "createdKind", ValueFunc: prometheus.FromLabelValue("kube_pod_info", "created_by_kind"), Type: sdkMetric.ATTRIBUTE, Optional: true},
 			{Name: "createdBy", ValueFunc: prometheus.FromLabelValue("kube_pod_info", "created_by_name"), Type: sdkMetric.ATTRIBUTE, Optional: true},
 			{Name: "nodeIP", ValueFunc: prometheus.FromLabelValue("kube_pod_info", "host_ip"), Type: sdkMetric.ATTRIBUTE},
 			{Name: "namespace", ValueFunc: prometheus.FromLabelValue("kube_pod_info", "namespace"), Type: sdkMetric.ATTRIBUTE},
 			{Name: "namespaceName", ValueFunc: prometheus.FromLabelValue("kube_pod_info", "namespace"), Type: sdkMetric.ATTRIBUTE},
 			{Name: "nodeName", ValueFunc: prometheus.FromLabelValue("kube_pod_info", "node"), Type: sdkMetric.ATTRIBUTE},
 			{Name: "podName", ValueFunc: prometheus.FromLabelValue("kube_pod_info", "pod"), Type: sdkMetric.ATTRIBUTE},
-			{Name: "isReady", ValueFunc: definition.Transform(prometheus.FromLabelValue("kube_pod_status_ready", "condition"), toNumericBoolean), Type: sdkMetric.GAUGE},
+			// we are adding as default `false` since all ksm pods used refers to pending pods due to the IDGenerator.
+			{Name: "isReady", ValueFunc: definition.Transform(fetchWithDefault(prometheus.FromLabelValue("kube_pod_status_ready", "condition"), "false"), toNumericBoolean), Type: sdkMetric.GAUGE},
```

Comment:
This could be replace by something like:
```go
{Name: "isReady", ValueFunc: func(_, _ string, _ definition.RawGroups) (definition.FetchedValue, error) { return 0, nil }, Type: sdkMetric.GAUGE},
```

I opted for the actual one just to follow the same approach, but I am doubting among the two options

Reply:
I think I'm missing something. The alternative you proposed would not do the same, wold it? It always returns regardless of the `"condition"` label value which is taken into account in the solution you chose... Sorry for may possible lack of context here.

Reply:
I think I'm missing something. The alternative you proposed would not do the same, wold it? It always returns regardless of the `"condition"` label value which is taken into account in the solution you chose... Sorry for may possible lack of context here.

Reply:
The code is quite confusing.
The metric should be never present since we are fetching pod data from KSM merely if the pod is `Pending`. You can check the `IDGenerator` few lines above

Reply:
Now I get the reason why both are equivalent, thanks for the explanation!